### ISSUE TITLE
Forbid returning too long tuples

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1731]
# Rule request

## Thesis

Forbid returning a too long constant-length tuple  from a function:

```python
# bad
def hello():
  return 1, 2, 3, 4
```

When there are more than 2 values must be returned, use [dataclasses.dataclass](https://docs.python.org/3/library/dataclasses.html), [typing.NamedTuple](https://docs.python.org/3/library/typing.html#typing.NamedTuple), or [attrs](https://www.attrs.org/en/stable/) instead.

## Reasoning

Code readability and maintainability: the caller code must know exact order of all components which is easy to mess up. A real-world example:

```python
time, phase, spacing, size, version, error, series_type = read_result_log()
```

:weary:

**Comment by sobolevn** [comment]
We already have:
https://github.com/wemake-services/wemake-python-styleguide/blob/2214832b74098b15ddf0eed55c92ab8265150d01/wemake_python_styleguide/violations/complexity.py#L883

and

https://github.com/wemake-services/wemake-python-styleguide/blob/2214832b74098b15ddf0eed55c92ab8265150d01/wemake_python_styleguide/violations/complexity.py#L1153

Maybe it is a good idea to simplify them into a single rule.

On the other hand, I like to define long immutable sequences as tuples. I guess that this should be allowed.

**Comment by orsinium** [comment]
Hmm. I tried the snippet above on [flakehell.orsinium.dev](https://flakehell.orsinium.dev/) and it hasn't found anything :thinking: I'm wondering what I did wrong.

![image](https://user-images.githubusercontent.com/9638362/99410661-091d2100-28f3-11eb-9b52-7f2298c12238.png)

**Comment by orsinium** [comment]
Oh, I see, it's only about `yield`, not `return`

> On the other hand, I like to define long immutable sequences as tuples. I guess that this should be allowed.

Sure. This is why I said "constant-length". When you use a tuple as an immutable list (when the length is unknown in advance), it is fine.

**Comment by uhbif19** [comment]
@sobolevn So am I understanding correctly, that the issue realization must just add full `TooLongYieldTupleViolation` analog but for `return`?

**Comment by orsinium** [comment]
Exactly! I don't see a reason why `return` is any different from `yield` from this perspective :)

**Comment by uhbif19** [comment]
Please assign me, I am finishing PR for this task.
